backend/utils/bulk_import.py

```
@@ -0,0 +1,448 @@
+    try:
+        if result.success_count > 0:
+            db.session.commit()
+            logger.info(f"Successfully imported {result.success_count} tools")
+        else:
+            db.session.rollback()
+    except Exception as e:
+        db.session.rollback()
+        logger.error(f"Error committing tool imports: {str(e)}")
+        # Convert all successes to errors
+        for item in result.created_items:
+            result.add_error(0, item['data'], f"Database commit failed: {str(e)}")
```

Flake8 (7.2.0)

[error] line too long (82 > 79 characters)

(E501)